src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -41,4 +43,19 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
+
+    @Override
+    public SizeResponse restoreSize(UUID sizeId){
+        validateSizeForRestoration(sizeId);
+        Size size = findSizeByIdOrThrowException(sizeId);
+        size.setDeleted(false);
+        return SizeMapper.toSizeResponse(sizeRepository.save(size));
+    }
+
+    public void validateSizeForRestoration(UUID sizeId){
+        Size size = findSizeByIdOrThrowException(sizeId);
+        if(!size.isDeleted()){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + sizeId + " is not deleted." );
+        }
+    }
```

Comment:
Siguiendo la corrección que le hicimos al Nico (y que no aplicamos en el restaurar producto porque somos re inteligentes) esto podría optimizarse para no pegarle dos veces a la base de datos

```suggestion
    public void validateSizeForRestoration(Size size) {
        if (!size.isDeleted()) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Size with id " + sizeId + " is not deleted." );
        }
    }
```